ReSharper.FSharp/src/FSharp.TypeProvidersProtocol/src/Utils/ProvidedTypesComparer.cs

```diff
@@ -9,10 +9,64 @@ private ProvidedTypesComparer()
     {
     }
 
-    public bool Equals(ProvidedType x, ProvidedType y) =>
-      x?.FullName == y?.FullName && x?.Assembly.FullName == y?.Assembly.FullName;
+    private static bool GetFullNameFast(ProvidedType x, out string fullName)
+    {
+      try
+      {
+        fullName = x.FullName;
+        return true;
+      }
+      catch
```

Comment:
Decided to at least leave a comment here (since we log the exceptions somewhere else).